lib/ansible/utils/encrypt.py

```diff
@@ -146,7 +170,7 @@ def _hash(self, secret, salt, rounds, ident):
         # crypt.crypt on Python < 3.9 returns None if it cannot parse saltstring
         # On Python >= 3.9, it throws OSError.
         try:
-            result = crypt.crypt(secret, saltstring)
+            result = to_text(crypt(to_bytes(secret), to_bytes(saltstring)))
             orig_exc = None
         except OSError as e:
```

Comment:
This `OSError` is incorrect with ctypes, will need to be adjusted.  Likely as `ValueError`, will need to confirm.